src/coreComponents/physicsSolvers/wavePropagation/ElasticFirstOrderWaveEquationSEMKernel.hpp

```diff
@@ -0,0 +1,636 @@
+   * @param[in] numFacesPerElem number of face on an element
+   * @param[in] X coordinates of the nodes
+   * @param[in] elemGhostRank array containing the ghost rank
+   * @param[in] elemsToNodes map from element to nodes
+   * @param[in] elemsToFaces map from element to faces
+   * @param[in] elemCenter coordinates of the element centers
+   * @param[in] faceNormal array containing the normal of all faces
+   * @param[in] faceCenter array containing the center of all faces
+   * @param[in] sourceCoordinates coordinates of the source terms
+   * @param[in] receiverCoordinates coordinates of the receiver terms
+   * @param[in] dt time-step
+   * @param[in] timeSourceFrequency Peak frequency of the source
+   * @param[in] rickerOrder Order of the Ricker wavelet
+   * @param[out] sourceIsLocal flag indicating whether the source is local or not
+   * @param[out] sourceNodeIds indices of the nodes of the element where the source is located
+   * @param[out] sourceConstantsx constant part of the source terms in x-direction
+   * @param[out] sourceConstantsy constant part of the source terms in y-direction
+   * @param[out] sourceConstantsz constant part of the source terms in z-direction
+   * @param[out] receiverIsLocal flag indicating whether the receiver is local or not
+   * @param[out] receiverNodeIds indices of the nodes of the element where the receiver is located
+   * @param[out] receiverNodeConstants constant part of the receiver term
+   * @param[out] sourceValue array containing the value of the time dependent source (Ricker for e.g)
+   */
+  template< typename EXEC_POLICY, typename FE_TYPE >
+  static void
+  launch( localIndex const size,
+          localIndex const numNodesPerElem,
+          localIndex const numFacesPerElem,
+          arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const X,
+          arrayView1d< integer const > const elemGhostRank,
+          arrayView2d< localIndex const, cells::NODE_MAP_USD > const & elemsToNodes,
+          arrayView2d< localIndex const > const elemsToFaces,
+          arrayView2d< real64 const > const & elemCenter,
+          arrayView2d< real64 const > const faceNormal,
+          arrayView2d< real64 const > const faceCenter,
+          arrayView2d< real64 const > const sourceCoordinates,
+          arrayView1d< localIndex > const sourceIsAccessible,
+          arrayView1d< localIndex > const sourceElem,
+          arrayView2d< localIndex > const sourceNodeIds,
+          arrayView2d< real64 > const sourceConstants,
+          arrayView2d< real64 const > const receiverCoordinates,
+          arrayView1d< localIndex > const receiverIsLocal,
+          arrayView1d< localIndex > const rcvElem,
+          arrayView2d< localIndex > const receiverNodeIds,
+          arrayView2d< real64 > const receiverConstants,
+          arrayView2d< real32 > const sourceValue,
+          real64 const dt,
+          real32 const timeSourceFrequency,
+          localIndex const rickerOrder )
```

Comment:
Maybe it is worth merging this function with `PrecomputeSourceAndReceiverKernel` from `AcousticFirstOrderWaveEquationSEMKernel.hpp` (and the other kernel files) because it is essentially the same kernel duplicated fourth times, unless I missed something?

Reply:
Yes François I agree. 
In fact 3 of the 4 propagators (including the one in this PR) use the exact same function. 
However, as it will modify calls inside nearly all solvers, we think that we can keep it that way for the framework of this PR, in order to merge it (it started to be here for a long time...) and then we will do a refactoring PR which will adress this (and maybe other somilar points).